src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -222,12 +220,71 @@ MultiPhysicalPlanCreate(MultiTreeRoot *multiTree)
 	multiPlan->workerJob = workerJob;
 	multiPlan->masterQuery = masterQuery;
 	multiPlan->masterTableName = jobSchemaName->data;
+	multiPlan->routerExecutable = MultiPlanRouterExecutable(multiPlan);
 
 	return multiPlan;
 }
 
 
 /*
+ * MultiPlanRouterExecutable returns true if the input multiPlan is
+ * router executable.
+ *
+ * Note that all the multi plans that are created by router planner are
+ * already router executable. Thus, this function should only be called
+ * for multi plans that are not generated by router planner.
+ */
+static bool
+MultiPlanRouterExecutable(MultiPlan *multiPlan)
+{
+	Query *masterQuery = multiPlan->masterQuery;
+	Job *job = multiPlan->workerJob;
+	List *workerTaskList = job->taskList;
+	int taskCount = list_length(workerTaskList);
+	int dependedJobCount = list_length(job->dependedJobList);
+	bool masterQueryHasAggregates = false;
+
+	/* router executor cannot execute SELECT queries that hit more than one shard */
+	if (taskCount != 1)
+	{
+		return false;
+	}
+
+	/* router executor cannot execute repartition jobs */
+	if (dependedJobCount > 0)
+	{
+		return false;
+	}
+
+	/* router executor cannot execute queries with order by */
+	if (masterQuery != NULL && list_length(masterQuery->sortClause) > 0)
+	{
+		return false;
+	}
+
+	/*
+	 * Router executor cannot execute queries with aggregates.
```

Comment:
That's not really true... Maybe rephrase this to say that router doesn't run the master query, thus aggregation/sorting on the master wouldn't be executed, and hence may not be present?